src/components/engagement/pnp-extractor.service.ts

```diff
@@ -16,12 +16,47 @@ export class PnpExtractor {
     @Logger('pnp:extractor') private readonly logger: ILogger
   ) {}
 
+  private year = 0;
+  private quarter = 0;
+
+  private getYearAndQuarterFromFileName(fileName: string) {
+    // first check for 2021, 2020, 2019
+    // the reason I did this is b/c some file names had multiple sets of 4 numbers (2020 and 2003 for instance)
+    const reg2021 = /2021/;
+    const reg2020 = /2020/;
+    const reg2019 = /2019/;
+    // regex for year between 1900 and 2099
+    const yearReg = /19\d{2}|20\d{2}/;
+    const fyReg = /fy19|fy20|fy21/i;
+    const quarterReg = /q[1-4]/i;
+
+    const year = reg2021.exec(fileName)
+      ? Number(reg2021.exec(fileName))
+      : reg2020.exec(fileName)
+      ? Number(reg2020.exec(fileName))
+      : reg2019.exec(fileName)
+      ? Number(reg2019.exec(fileName))
+      : yearReg.exec(fileName)
+      ? Number(yearReg.exec(fileName))
+      : fyReg.exec(fileName)
+      ? Number('20' + fyReg.exec(fileName)![0].toLowerCase().replace('fy', ''))
+      : 0;
+
+    const quarter = quarterReg.exec(fileName)
+      ? Number(quarterReg.exec(fileName)![0].toLowerCase().replace('q', ''))
+      : 0;
+
+    this.year = year;
+    this.quarter = quarter;
+  }
+
   async extract(
     input: CreateDefinedFileVersionInput,
     session: Session
   ): Promise<PnpData | null> {
     const file = await this.files.getFileVersion(input.uploadId, session);
     const pnp = await this.downloadWorkbook(file);
+    this.getYearAndQuarterFromFileName(input.name!);
```

Comment:
input.name is optional and defaults to the file's name. Use it instead.
```suggestion
    this.getYearAndQuarterFromFileName(file.name);
```